packages/legend-application-query/src/components/__tests__/QueryEditorQueryActions.test.tsx

```diff
@@ -225,3 +229,183 @@ test(
     ).toBe(false);
   },
 );
+
+test(
+  integrationTest("Query header actions are disabled if query can't be built"),
+  async () => {
+    const { renderResult, queryBuilderState } = await TEST__setUpQueryEditor(
+      TEST__provideMockedQueryEditorStore(),
+      TEST_DATA__ResultState_entities,
+      stub_RawLambda(),
+      'execution::RelationalMapping',
+      'execution::Runtime',
+      TEST_DATA__modelCoverageAnalysisResult,
+    );
+    const _class = 'model::Firm';
+
+    const _modelClass =
+      queryBuilderState.graphManagerState.graph.getClass(_class);
+
+    await act(async () => {
+      queryBuilderState.changeClass(_modelClass);
+    });
+
+    const filterPanel = await waitFor(() =>
+      renderResult.getByTestId(
+        QUERY_BUILDER_TEST_ID.QUERY_BUILDER_FILTER_PANEL,
+      ),
+    );
+    const tdsProjectionPanel = await waitFor(() =>
+      renderResult.getByTestId(
+        QUERY_BUILDER_TEST_ID.QUERY_BUILDER_TDS_PROJECTION,
+      ),
+    );
+    const explorerPanel = await waitFor(() =>
+      renderResult.getByTestId(QUERY_BUILDER_TEST_ID.QUERY_BUILDER_EXPLORER),
+    );
+
+    // Drag and drop
+    const tdsProjectionDropZone = await waitFor(() =>
+      getByText(tdsProjectionPanel, 'Add a projection column'),
+    );
+    const filterDropZone = await waitFor(() =>
+      getByText(filterPanel, 'Add a filter condition'),
+    );
+    const dragSource = await waitFor(() =>
+      getByText(explorerPanel, 'Legal Name'),
+    );
+    await dragAndDrop(
+      dragSource,
+      tdsProjectionDropZone,
+      tdsProjectionPanel,
+      'Add a projection column',
+    );
+    await dragAndDrop(
+      dragSource,
+      filterDropZone,
+      filterPanel,
+      'Add a filter condition',
+    );
+    await waitFor(() => getByText(filterPanel, 'Legal Name'));
+    await waitFor(() => getByText(filterPanel, 'is'));
+    await waitFor(() => getByDisplayValue(filterPanel, ''));
+
+    // Verify action buttons are disabled properly
+    const queryActionsPanel = await renderResult.findByTestId(
+      QUERY_EDITOR_TEST_ID.QUERY_EDITOR_ACTIONS,
+    );
+    expect(
+      getByRole(queryActionsPanel, 'button', {
+        name: 'Load Query',
+      }).hasAttribute('disabled'),
+    ).toBe(false);
+    expect(
+      getByRole(queryActionsPanel, 'button', {
+        name: 'New Query',
+      }).hasAttribute('disabled'),
+    ).toBe(false);
+    expect(
+      getByRole(queryActionsPanel, 'button', { name: 'Save' }).hasAttribute(
+        'disabled',
+      ),
+    ).toBe(true);
+    expect(
+      getByRole(queryActionsPanel, 'button', {
+        name: 'Save As...',
+      }).hasAttribute('disabled'),
+    ).toBe(true);
+  },
```

Comment:
would you mind also testing the issue-count-badge you added?

Reply:
Done